griptape/drivers/prompt/openai_chat_prompt_driver.py

```diff
@@ -89,13 +120,35 @@ def try_stream(self, prompt_stack: PromptStack) -> Iterator[TextArtifact]:
         for chunk in result:
             if len(chunk.choices) == 1:
                 delta = chunk.choices[0].delta
-            else:
-                raise Exception("Completion with more than one choice is not supported yet.")
+                tool_call_deltas = delta.tool_calls
+
+                if tool_call_deltas:
+                    if len(tool_call_deltas) == 1:
+                        tool_call_delta = tool_call_deltas[0]
+
+                        if tool_call_delta.id is not None:
+                            name, path = tool_call_delta.function.name.split("-")
+                            value = f"{name}-{path}"
+                        else:
+                            name, path = None, None
+                            value = tool_call_delta.function.arguments or ""
 
-            if delta.content is not None:
-                delta_content = delta.content
+                        yield ActionChunkArtifact(
+                            value=value,
+                            index=tool_call_delta.index,
+                            tag=tool_call_delta.id,
+                            name=name,
+                            path=path,
+                            partial_input=tool_call_delta.function.arguments,
+                        )
+                    else:
+                        raise Exception("Streaming completion with more than one tool call delta is not supported yet.")
+                else:
+                    content_delta = delta.content or ""
```

Comment:
Is it ever possible for both `delta.content` and `delta.tool_calls` to be present / non-None?